### DISCUSSION TITLE
IBus for Plover

### DISCUSSION THREAD

**Discussion by user202729** [opensteno/plover#1390] [Show and tell]
... just made a plugin. (https://github.com/user202729/plover-ibus & https://github.com/user202729/ibus-plover-engine)

IBus is not as well-documented as I expected (or I didn't figure out where to read), so it took quite a long time. (and the code might not be the cleanest way to achieve the task. I need an additional socket connection)

Besides you have to install the plugin manually, although I guess it should be possible to do it automatically somehow...

At least I got something to work, and it does fix the Unicode output issues/race condition issues.

There are new issues however (have to figure out how to send `\n` or other key combo properly through IBus, and send manual backspace when `delete_surrounding_text` doesn't work)

------

Related: https://github.com/openstenoproject/plover/discussions/1426 (fix the missing output issue by delaying selectively *by focused application classes*, and some other Unicode output issues by paste from clipboard etc.)

**Comment by antoniusf** [comment]
Omg, I've only just seen this! Thanks so much for all of your work (the uinput plugin too) in making plover get along better with wayland-based compositors.
